examples/ExactlyOnceBasic/Program.cs

```diff
@@ -0,0 +1,680 @@
+// Copyright 2022 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using Confluent.Kafka;
+using Confluent.Kafka.Admin;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.IO;
+
+
+namespace Confluent.Kafka.Examples.ExactlyOnceBasic
+{
+    /// <summary>
+    ///     Main class of this EOS producer example.
+    /// </summary>
+    public class Program
+    {
+
+        /// <summary>
+        ///     Logs provided <paramref name="message"/>.
+        /// </summary>
+        public static void Log(string message)
+        {
+            Console.Error.WriteLine($"{DateTimeOffset.Now.ToUnixTimeSeconds()}|EXAMPLE|{message}");
+        }
+
+        private static async Task CreateTopic(IAdminClient adminClient, string name, int partitions)
+        {
+
+            try
+            {
+                await adminClient.CreateTopicsAsync(new List<TopicSpecification>
+                {
+                    new TopicSpecification
+                    {
+                        Name = name,
+                        NumPartitions = partitions
+                    }
+                });
+                Log($"Created topic {name}");
+            }
+            catch (KafkaException)
+            {
+                // Continue if topic already exists
+                Log($"Topic {name} already exists");
+            }
+        }
+
+        /// <summary>
+        ///     Main method.
+        /// </summary>
+        public static async Task Main(string[] args)
+        {
+            CancellationTokenSource cancelled = new CancellationTokenSource();
+            CancellationTokenSource completed = new CancellationTokenSource();
+            Console.CancelKeyPress += (_, e) =>
+            {
+                e.Cancel = true; // prevent the process from terminating.
+                cancelled.Cancel();
+            };
+            AppDomain.CurrentDomain.ProcessExit += (_, e) =>
```

Comment:
seems too fancy to use AppDomain. Why not just wait on the cancellation token at the end of Main? `cts.Token.WaitHandle.WaitOne();`

Reply:
I've added it because, you don't, it handles SIGINT correctly but not SIGTERM and doesn't exit gracefully. It happens the same if you don't wait for completedCTS.